src/MagnumPlugins/KtxImporter/KtxImporter.cpp

```diff
@@ -0,0 +1,891 @@
+            /* Length without padding */
+            const UnsignedInt length = *reinterpret_cast<const UnsignedInt*>(keyValueData.suffix(current).data());
+            Utility::Endianness::littleEndianInPlace(length);
+            current += sizeof(length);
+
+            if(current + length < keyValueData.size()) {
+                const Containers::StringView entry{keyValueData.suffix(current).prefix(length)};
+                const Containers::Array3<Containers::StringView> split = entry.partition('\0');
+                const auto key = split[0];
+                const auto value = split[2];
+
+                if(key.isEmpty() || value.isEmpty())
+                    Warning{} << "Trade::KtxImporter::openData(): invalid key/value entry, skipping";                    
+                else if(keyValueMap.count(key) > 0)
+                    Warning{} << "Trade::KtxImporter::openData(): key" << key << "already set, skipping";
+                else
+                    keyValueMap[key] = value;
+            }
+            /* Length value is dword-aligned, guaranteed for the first length
+               by the file layout */
+            current += (length + 3)/4*4;
+        }
+    }
+
+    using namespace Containers::Literals;
+
+    /* Read image orientation so we can flip if needed.
+
+       l/r = left/right
+       u/d = up/down
+       o/i = out of/into screen
+
+       The spec strongly recommends defaulting to rdi, Magnum/GL expects ruo. */
+    {
+        constexpr auto targetOrientation = "ruo"_s;
+
+        bool useDefaultOrientation = true;
+        const auto found = keyValueMap.find("KTXorientation"_s);
+        if(found != keyValueMap.end()) {
+            const Containers::StringView orientation{found->second};
+            if(orientation.size() >= f->numDimensions) {
+                constexpr Containers::StringView validOrientations[3]{"rl"_s, "du"_s, "io"_s};
+                for(UnsignedByte i = 0; i != f->numDimensions; ++i) {
+                    useDefaultOrientation = !validOrientations[i].contains(orientation[i]);
+                    if(useDefaultOrientation) break;
+                    f->flip.set(i, orientation[i] != targetOrientation[i]);
+                }
+            }
+        }
+
+        if(useDefaultOrientation) {
+            constexpr Containers::StringView defaultDirections[3]{
+                "right"_s, "down"_s, "forward"_s
+            };
+            Warning{} << "Trade::KtxImporter::openData(): missing or invalid "
+                "orientation, assuming" << ", "_s.join(Containers::arrayView(defaultDirections).prefix(f->numDimensions));
+
+            constexpr auto defaultOrientation = "rdi"_s;
+            const BoolVector3 flip = Math::notEqual(Math::Vector3<char>::from(defaultOrientation.data()),
+                Math::Vector3<char>::from(targetOrientation.data()));
+            for(UnsignedByte i = 0; i != f->numDimensions; ++i)
+                f->flip.set(i, flip[i]);
```

Comment:
I'm afraid that still sets it to `true` for higher dimensions that should be ignored. For e.g. a 1D image without metadata, `flip` would become `{false, true, true}`. This makes `any()`/`none()` useless, which is used right below. I added a comment to clarify since I couldn't find a code golf solution that didn't make the code unreadable.